client.go

```diff
@@ -689,6 +733,8 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 
 		if c.RequestLogHook != nil {
 			switch v := logger.(type) {
+			case ContextLogger:
+				c.RequestLogHook(contextLogger{v}, req.Request, i)
```

Comment:
I think you should pass `ctx` to `contextLogger`.

Reply:
In `RequestLogHook`, you accept the `*http.Request` parameter, and you can get the Context from it. If the user overrides `RequestLogHook`, you can get the Context in this way. However, if you put it in `contextLogger`, it cannot be retrieved and converted externally, because it is not external in design

Reply:
Yes, `ctx` passed to `contextLogger` does not have be public, but it should be passed to `InfoContext` in `Printf`.

Reply:
You're right